### ISSUE TITLE
DllPlugin does not write manifest when used with webpack-dev-server

### ISSUE THREAD

**Issue by robrichard** [webpack/webpack#6052]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Currently the DLL plugin (via LibManifestPlugin) [writes the manifest using `compiler.outputFileSystem`
](https://github.com/webpack/webpack/blob/master/lib/LibManifestPlugin.js#L55). When this is run with `webpack-dev-server` the `outputFileSystem` is the in-memory filesystem. It is not accessible from another webpack project's `DllReferencePlugin` unless it is first downloaded from `webpack-dev-server`.

**If the current behavior is a bug, please provide the steps to reproduce.**
Run any webpack config that generates a DLL with webpack-dev-server and the manifest file is not written to disk.

**What is the expected behavior?**
`DllPlugin` should always write to the real filesystem.

**If this is a feature request, what is motivation or use case for changing the behavior?**
This will make it much easier to do active development on code that is in a webpack DLL.

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
node@6.11.2, webpack@3.9.1, macOS@10.13.1

**Comment by sokra** [comment]
> DllPlugin should always write to the real filesystem.

or

DllReferecePlugins should be able to read the manifest from the memory filesystem.

**Comment by robrichard** [comment]
@sokra yes that would also be a solution to this problem. Which approach is preferable? I'd be happy to work on a PR if you can point me in the right direction.

**Comment by sokra** [comment]
> DllReferencePlugin should be able to read the manifest from the memory filesystem.

This would be preferred.

You can solve this by decorating the inputFileSystem with a filesystem reading from two filesystems (the real one and the in-memory one).

**Comment by robrichard** [comment]
What about an option to pass a manifest URL to DllReferencePlugin? Then it can download it over HTTP from the dev server?

**Comment by fortinmike** [comment]
I'm encountering this bug too.

Is there a recommended workaround that doesn't require modifying webpack?
